cranelift/codegen/src/opts/algebraic.isle

```diff
@@ -32,6 +32,9 @@
                       x))
       (ineg ty x))
 
+;; x-x == 0.
+(rule (simplify (isub ty x x)) (subsume (iconst ty (imm64 0))))
```

Comment:
This works great for all types except `$I128`, which for practical reasons (size of operator struct) isn't supported by `iconst`; so we need to guard this rule by a `fits_in_64` pattern, as for `x ^ x` on line 113 below.

Reply:
Good catch, thanks!